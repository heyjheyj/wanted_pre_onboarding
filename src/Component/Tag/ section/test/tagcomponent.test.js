import React from "react";
import TestRenderer from "react-test-renderer";
import TagComponent from "../tagComponent";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Tag Component", () => {
  let deleteTag = jest.fn();
  let tag = { id: 1, text: "Hello" };
  let button;

  it("renders", () => {
    const component = TestRenderer.create(
      <TagComponent deleteTag={deleteTag} tag={tag} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("Delete Tag", () => {
    beforeEach(() => {
      render(<TagComponent deleteTag={deleteTag} tag={tag} />);
    });

    it("calls deleteTag when button is clicked", () => {
      button = screen.getByTestId("test-deletebutton1");
      userEvent.click(button);

      expect(deleteTag).toHaveBeenCalledTimes(1);
      expect(deleteTag).toHaveBeenCalledWith(tag);
    });
  });
});
