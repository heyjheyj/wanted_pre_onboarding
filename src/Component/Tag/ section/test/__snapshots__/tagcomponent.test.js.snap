// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component renders 1`] = `
<li
  className="tagcomponent"
  data-testid="test-tag1"
>
  <span
    data-testid="test-text1"
  >
    Hello
  </span>
  <button
    data-testid="test-deletebutton1"
    onClick={[Function]}
  >
    x
  </button>
</li>
`;
